src/main/scala/nl/biopet/bioconda/BiocondaUtils.scala

```diff
@@ -0,0 +1,209 @@
+      Some(jar.openStream().sha256.hex)
+    } catch {
+      case e: java.io.FileNotFoundException => None
+    }
+  }
+
+  def getSourceUrl(tag: TagName, repo: GHRepository): Option[String] = {
+    val repoName = repo.getName()
+    val repoOwner = repo.getOwnerName()
+    // Disable authentication. These jars should be accessible for non authenticated users.
+    val noAuthRepo =
+      GitHub.connectAnonymously().getUser(repoOwner).getRepository(repoName)
+    val releaseList = noAuthRepo.listReleases().asList()
+    val releases =
+      JavaConverters.collectionAsScalaIterable(releaseList).toList
+    val currentRelease = releases.find(x => x.getTagName == tag)
+    if (currentRelease.isEmpty) {
+      throw new Exception(
+        s"'$tag' tag not present on release page. Please release on github before publishing to bioconda.")
+    }
+    val assets = JavaConverters
+      .collectionAsScalaIterable(
+        currentRelease.getOrElse(new GHRelease).getAssets)
+      .toList
+    val releaseJar =
+      // Finds all jars. This assumes only one jar is released.
+      assets.find(
+        x =>
+          x.getBrowserDownloadUrl.endsWith(".jar") &&
+            !x.getBrowserDownloadUrl().endsWith("javadoc.jar") &&
+            !x.getBrowserDownloadUrl.endsWith("sources.jar")
+      )
+    if (releaseJar.isEmpty) { None } else
+      Some(releaseJar.getOrElse(new GHAsset).getBrowserDownloadUrl)
+  }
+  def branchExists(branch: String,
+                   repo: File,
+                   git: GitRunner,
+                   log: ManagedLogger,
+                   remotes: Boolean = false): Boolean = {
+    // TODO: Find a git command that just returns branches as a list. (Without * in front of the branch you are on)
+
+    // Without "--no-color" scala doesn't match the strings properly. Color matters in string comparison!
+    val branchList: Array[String] = {
+      if (remotes) {
+        git.apply("branch", "-a", "--no-color")(repo, log).split("\\n")
+      } else {
+        git.apply("branch", "--no-color")(repo, log).split("\\n")
+      }
+    }
+    val branches = new ListBuffer[String]
+
+    // For each branch
+    // Remove that annonoying *
+    // Split on / and get the last item(remotes/origin/branch) => branch
+    // Trim away all spaces
+    branchList.foreach(x =>
+      branches.append(x.replaceFirst("\\*", "").split("/").last.trim()))
+    branches.toList.contains(branch)
+
+  }
+  def getVersionFromYaml(metaYaml: File): String = {
+    val versionRegex: Regex = "version:.([0-9\\.]+)".r
+    val yaml = Source.fromFile(metaYaml).getLines().mkString
+    val matches = versionRegex.findAllIn(yaml).matchData
+    if (matches.isEmpty) {
+      throw new Exception(s"No version found in: ${metaYaml.getPath}")
+    }
+    val version = matches.toList.head.group(1)
+    version
+  }
+
+  def dockerInstalled(log: ManagedLogger, path: Option[String] = None): Unit = {
+    val testCommand = "docker version -f '{{.Client.Version}}'"
```

Comment:
There is a special package inside CommonUtils that deals with sys calls, also async is required. This will reduce the code here.

Reply:
This should also be rechecked again, must be possible now with utils in 2.12